packages/node-cli/src/commands/CreateProjectCommandGenerator.js

```diff
@@ -221,104 +224,115 @@ module.exports = class CreateProjectCommandGenerator extends BaseCommandGenerato
 	}
 
 	async _executeAction(answers) {
-		const projectFolderName = answers[COMMAND_ANSWERS.PROJECT_FOLDER_NAME];
-		const projectAbsolutePath = answers[COMMAND_OPTIONS.PARENT_DIRECTORY];
-		const manifestFilePath = path.join(
-			projectAbsolutePath,
-			SOURCE_FOLDER,
-			ApplicationConstants.FILES.MANIFEST_XML
-		);
+		try {
+			const projectFolderName = answers[COMMAND_ANSWERS.PROJECT_FOLDER_NAME];
+			const projectAbsolutePath = answers[COMMAND_OPTIONS.PARENT_DIRECTORY];
+			const manifestFilePath = path.join(
+				projectAbsolutePath,
+				SOURCE_FOLDER,
+				ApplicationConstants.FILES.MANIFEST_XML
+			);
 
-		const validationErrors = this._validateParams(answers);
+			const validationErrors = this._validateParams(answers);
 
-		if (validationErrors.length > 0) {
-			throwValidationException(validationErrors, false, this._commandMetadata);
-		}
+			if (validationErrors.length > 0) {
+				throwValidationException(validationErrors, false, this._commandMetadata);
+			}
+
+			const params = {
+				//Enclose in double quotes to also support project names with spaces
+				parentdirectory: CommandUtils.quoteString(projectAbsolutePath),
+				type: answers[COMMAND_OPTIONS.TYPE],
+				projectname: SOURCE_FOLDER,
+				...(answers[COMMAND_OPTIONS.OVERWRITE] && { overwrite: '' }),
+				...(answers[COMMAND_OPTIONS.TYPE] === ApplicationConstants.PROJECT_SUITEAPP && {
+					publisherid: answers[COMMAND_OPTIONS.PUBLISHER_ID],
+					projectid: answers[COMMAND_OPTIONS.PROJECT_ID],
+					projectversion: answers[COMMAND_OPTIONS.PROJECT_VERSION],
+				}),
+			};
+
+			this._fileSystemService.createFolder(this._executionPath, projectFolderName);
+
+			const actionCreateProject = new Promise(async (resolve, reject) => {
```

Comment:
It would be a good idea to extract this `const actionCreateProject  = new Promise(...)` to another method so this one is cleaner

Reply:
Extreacted to method createProject